### DISCUSSION TITLE
Named rules rather than numbered?

### DISCUSSION THREAD

**Discussion by TonyGravagno** [zokugun/vscode-explicit-folding#119] [Issue Triage]
Looking at the debug output, rules are numbered using the zero-based index in which they are executed: "regex: 0 , 1, 2 ..."

What do you think about a "name" property in rules that will show in debug in addition to the index?

> regex: 5
> regex: 9 "curly brace block"
> regex: 7 "parentheses on same line"

```json
"explicitFolding.rules": {
 ...
    {
       "name": "curly brace block",
       "begin": "{",
       ...
    }
}
```

Bonus? : Add the group name from the new "include" feature:

> regex: 5 in "*"
> regex: 9 "curly brace block" in "simple-js"
> regex: 7 "parentheses on same line" in "simple-js"

**Comment by daiyam** [comment]
I don't know, let me sleep on it...

**Comment by TonyGravagno** [comment]
Good morning! 🤣 
I was looking at debug output yesterday and thinking about this feature again.

**Reply by daiyam** [reply]
I did sleep quite a bit :wink:

`name` is already available for `begin`/`end` rules (https://github.com/zokugun/vscode-explicit-folding/blob/master/docs/rules/begin-end.md#name, how it's used: https://github.com/zokugun/vscode-explicit-folding/discussions/60) but not for other rules. I should fix that.

The group name could be included to give:
`[curly brace block#simple-js] line: 77, offset: 0, type: BEGIN, match: ..., regex: 0`

**Reply by TonyGravagno** [reply]
> name is already available for begin/end rules ... but not for other rules. I should fix that.

I'm not sure if you're including beginRegex with begin/end, or if regex rules are "other". But 'name' doesn't seem to be used for beginRegex either. FYI / HTH / LOL

**Reply by daiyam** [reply]
There is 3 `begin` rules:
- `begin/end`
- `begin/continuation`
- `begin/while`

cf https://github.com/zokugun/vscode-explicit-folding/tree/master/docs/rules

`begin` and `beginRegex`, when naming a rule, are also synonymous since both are always supported.

**Comment by daiyam** [comment]
With https://github.com/zokugun/vscode-explicit-folding/pull/135, I get:
```
[main, #region, 3fold] line: 16, offset: 0, type: BEGIN, match: {{{, regex: 0
[main, #region, 3fold] line: 20, offset: 0, type: END, match: }}}, regex: 0
```

**Reply by TonyGravagno** [reply]
Pulling in all recent changes. Love it. Thanks! Will comment soon.

**Reply by TonyGravagno** [reply]
Nice!
<img width="1087" height="100" alt="image" src="https://github.com/user-attachments/assets/e684d777-0eec-47e5-889d-8b252e977133" />
